akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -82,7 +82,7 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
     override def toString = "GraphStageLogicSpec.emitEmptyIterable"
   }
 
-  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
+  private case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```

Comment:
Yes, those are trickier. Would be nice if we could just suppress those warnings.